apps/frontend/src/pages/project/libs/components/setup-analytics-modal/setup-analytics-modal.tsx

```diff
@@ -29,19 +30,33 @@ const SetupAnalyticsModal = ({
 	project,
 }: Properties): JSX.Element => {
 	const dispatch = useAppDispatch();
-	const { control, errors, handleSubmit, handleValueSet } = useAppForm({
-		defaultValues: {
-			apiKey: project.apiKey ?? "",
-			projectId: project.id,
-		},
-	});
+
 	const { dataStatus } = useAppSelector(({ projectApiKeys }) => projectApiKeys);
+	const { authenticatedUser } = useAppSelector(({ auth }) => auth);
 
 	const hasProjectApiKey = project.apiKey !== null;
+	const hasAuthenticatedUser = authenticatedUser !== null;
+
 	const isGenerateButtonDisabled = dataStatus === DataStatus.PENDING;
 	const isCopyButtonDisabled =
 		!hasProjectApiKey || dataStatus === DataStatus.PENDING;
 
+	const script = useMemo<string>(
+		() =>
+			hasProjectApiKey && hasAuthenticatedUser
+				? `npx @git-fit/analytics@latest track ${project.apiKey as string} ${String(authenticatedUser.id)} <project-path>`
+				: "",
```

Comment:
Let's split it to some variables inside useMemo, it's a bit hard to read

Reply:
Is it okay?
```
	const script = useMemo<string>(() => {
		if (!hasProjectApiKey || !hasAuthenticatedUser) {
			return "";
		}

		const apiKey = project.apiKey as string;
		const userId = String(authenticatedUser.id);

		return `npx @git-fit/analytics@latest track ${apiKey} ${userId} <project-path>`;
	}, [hasProjectApiKey, hasAuthenticatedUser, project, authenticatedUser]);
```

Reply:
yes, looks good to me